azuredevops/utils/tfhelper/tfhelper.go

```
@@ -13,3 +16,58 @@ func DiffFuncSupressCaseSensitivity(k, old, new string, d *schema.ResourceData)
 	}
 	return false
 }
+
+func calcSecretHashKey(secretKey string) string {
+	return secretKey + "_hash"
+}
+
+// DiffFuncSupressSecretChanged is used to supress unneeded `apply` updates to a resource.
+//
+// It returns `true` when `new` appears to be the same value
+// as a previously stored and bcrypt'd value stored in state during a previous `apply`.
+// Relies on flatten/expand logic to help store that hash. See FlattenSecret, below.*/
+func DiffFuncSupressSecretChanged(k, old, new string, d *schema.ResourceData) bool {
+	memoKey := calcSecretHashKey(k)
+	memoValue := d.Get(memoKey).(string)
+
+	isUpdating, _, err := secretmemo.IsUpdating(new, memoValue)
+	isUnchanged := !isUpdating
+
+	if nil != err {
+		log.Printf("Change forced. Swallowing err while using secret hashing: %s", err)
+		return false
+	}
+
+	log.Printf("\nk: %s, old: %s, new: %s, memoKey: %s, memoValue: %s, isUnchanged: %t\n",
+		k, old, new, memoKey, memoValue, isUnchanged)
+	return isUnchanged
+}
+
+// HelpFlattenSecret is used to store a hashed secret value into `tfstate`
+func HelpFlattenSecret(d *schema.ResourceData, secretKey string) {
+	if !d.HasChange(secretKey) {
+		log.Printf("Secret key %s didn't get updated.", secretKey)
+		return
+	}
+	hashKey := calcSecretHashKey(secretKey)
+	newSecret := d.Get(secretKey).(string)
+	oldHash := d.Get(hashKey).(string)
+	_, newHash, err := secretmemo.IsUpdating(newSecret, oldHash)
+	if nil != err {
+		log.Printf("Swallowing err while using secret hashing: %s", err)
```

Can you elaborate on what the implications of swallowing this error are? When would this happen? Based on a brief grok, it seems to happen when the bcrypt fails -- but what "user scenario" does this map to?

---

Such an error would be unexpected (like env problems finding crypto libs or some such), but I can envision the stored hash being mangled, forgotten, etc. The `secretmemo.IsUpdating` impl handles more common problems by returning `true`.